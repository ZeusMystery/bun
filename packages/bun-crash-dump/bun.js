#!/usr/bin/env bun
await import('./main.js');
