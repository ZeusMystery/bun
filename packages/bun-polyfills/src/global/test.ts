import testPoly from '../modules/test.js';

//! Not a polyfill
// This file is used for preloading into the polyfills test runner, due to Jest's and by extension Bun's,
// quirky global scope pollution behavior.

for (const key of Object.keys(testPoly)) {
    Reflect.set(globalThis, key, Reflect.get(testPoly, key));
}
// Let's just not bother with concurrent tests for now, it'll work fine without. (Node's { concurrency: # } option is quirky)
Reflect.set(Reflect.get(globalThis, 'describe'), 'concurrent', testPoly.describe);
Reflect.set(Reflect.get(globalThis, 'test'), 'concurrent', testPoly.test);
Reflect.set(Reflect.get(globalThis, 'it'), 'concurrent', testPoly.it);