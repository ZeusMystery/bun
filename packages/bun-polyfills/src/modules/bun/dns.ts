import type { DNSLookup } from 'bun';
import dns from 'node:dns';

const dnsObj: typeof Bun.dns = {
    async lookup(hostname, options) {
        const opts = { verbatim: true, all: true, family: 0, hints: 0 } as Required<dns.LookupOptions>;
        if (options?.family) {
            if (options.family === 'IPv4') opts.family = 4;
            else if (options.family === 'IPv6') opts.family = 6;
            else if (options.family === 'any') opts.family = 0;
            else opts.family = options.family;
        }
        if (options?.flags) opts.hints = options.flags;
        opts.hints |= dns.V4MAPPED;
        const records = await dns.promises.lookup(hostname, opts) as dns.LookupAddress[];
        return await Promise.all(records.map(async r => {
            const record = r as DNSLookup;
            try {
                record.ttl = ((await dns.promises[`resolve${record.family}`](hostname, { ttl: true })) as dns.RecordWithTtl[])
                    .find(r => r.address === record.address)?.ttl ?? 0;
            } catch {
                record.ttl = 0;
            }
            return record;
        }));
    },
    // This has more properties but they're not documented on bun-types yet, oh well.
};

export default dnsObj;
