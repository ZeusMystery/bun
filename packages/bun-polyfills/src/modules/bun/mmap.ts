import fs from 'fs';
type mmapper = typeof import('mmap-utils')['default'];
type MapProtectionFlags = Parameters<mmapper['map']>[1];

let mmapper: mmapper | null = null;
try {
    // TS is having some trouble resolving these types properly, it thinks the module is on .default.default (???)
    mmapper = (await import('mmap-utils')).default as unknown as mmapper;
} catch {
    // Error will be thrown when mmap is used
}

//? The opts object may also support "size" and "offset" properties, but these are not documented in bun-types yet.
export const mmap: typeof Bun.mmap = (path, opts = {}): Uint8Array => {
    if (!mmapper) {
        const err = new Error('Bun.mmap is not available due to uninitialized mmapper dependency.');
        Error.captureStackTrace(err, mmap);
        throw err;
    }
    if (opts.shared === undefined) opts.shared = true;
    if (opts.sync === undefined) opts.sync = false;
    //? The sync option is ignored by Bun on MacOS and errors on Linux, so might as well just ignore it for now.
    //if (opts.sync) throw new NotImplementedError('Bun.mmap(..., { sync: true })', mmap);

    const fd = fs.openSync(path as fs.PathLike, 'r+');
    const size = fs.fstatSync(fd).size;
    return mmapper.map(
        size,
        <MapProtectionFlags>(mmapper.PROT_READ | mmapper.PROT_WRITE),
        opts.shared ? mmapper.MAP_SHARED : mmapper.MAP_PRIVATE,
        fd
    );
};
