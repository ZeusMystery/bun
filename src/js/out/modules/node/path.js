var E=function(i){var D=f({basename:i.basename.bind(i),dirname:i.dirname.bind(i),extname:i.extname.bind(i),format:i.format.bind(i),isAbsolute:i.isAbsolute.bind(i),join:i.join.bind(i),normalize:i.normalize.bind(i),parse:i.parse.bind(i),relative:i.relative.bind(i),resolve:i.resolve.bind(i),toNamespacedPath:i.toNamespacedPath.bind(i),sep:i.sep,delimiter:i.delimiter});return D.default=D,D},f=(i)=>Object.assign(Object.create(null),i),m=E(Bun._Path()),F=E(Bun._Path(!1)),g=E(Bun._Path(!0));m.win32=g;m.posix=F;var{basename:G,dirname:k,extname:H,format:q,isAbsolute:I,join:v,normalize:J,parse:y,relative:K,resolve:z,toNamespacedPath:L,sep:A,delimiter:N,__esModule:B}=m;m[Symbol.for("CommonJS")]=0;m.__esModule=!0;var C=m;export{g as win32,L as toNamespacedPath,A as sep,z as resolve,K as relative,F as posix,y as parse,J as normalize,v as join,I as isAbsolute,q as format,H as extname,k as dirname,N as delimiter,C as default,f as createModule,G as basename,B as __esModule};
