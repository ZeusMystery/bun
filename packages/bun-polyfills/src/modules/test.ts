import _dateFormat from 'dateformat';
const dateFormat = _dateFormat as unknown as typeof import('dateformat').default; // 10/10 types :D
import { expect as jestexpect } from 'expect';
import _jesteach from 'jest-each';
import extendedMatchers from 'jest-extended';
const jesteach = _jesteach.default as unknown as typeof import('jest-each').default; // bad types, again...
import * as jestmock from 'jest-mock';
import nodetest from 'node:test';
import chp from 'node:child_process';
import { promisify } from 'node:util';
import type { Mock } from 'bun:test';
const exec = promisify(chp.exec);

const bunmock: typeof import('bun:test')['mock'] = function mock(fn) {
    return jestmock.fn(fn);
};
bunmock.module = (id, factory) => { }; // TODO: Need to integrate this with the ESM loader somehow.
bunmock.restore = () => { }; // TODO

const bundescribe: typeof import('bun:test')['describe'] = (name, fn) => nodetest.describe(name, fn);
bundescribe.only = (name, fn) => nodetest.describe.only(name, fn);
bundescribe.todo = (name, fn) => nodetest.describe.todo(name, fn);
bundescribe.skip = (name, fn) => nodetest.describe.skip(name, fn);
bundescribe.skipIf = (condition) => condition ? nodetest.describe.skip : nodetest.describe;
bundescribe.if = (condition) => condition ? nodetest.describe : () => void 0;
bundescribe.each = (table: any) => {
    return (title: string, suite: AnyFunction) => jesteach(table).describe(title, suite);
};

jestexpect.extend(extendedMatchers);
const bunExpect = jestexpect as unknown as typeof import('bun:test')['expect'];
bunExpect.unreachable = (msg) => {
    if (msg instanceof Error) throw msg;
    else throw new Error(msg ?? 'Unreachable code reached');
};

const bunit: typeof import('bun:test')['it'] = (name, fn) => {
    nodetest.it(name, fn.length ? (t, done) => void fn(done) : async (t) => void await (fn as () => Promise<void>)());
};
bunit.only = (name, fn) => nodetest.only(name, fn.length ? (t, done) => void fn(done) : async (t) => void await (fn as () => Promise<void>)());
bunit.todo = (name, fn) => nodetest.todo(name, fn?.length ? (t, done) => void fn?.(done) : async (t) => void await (fn as () => Promise<void>)());
bunit.skip = (name, fn) => nodetest.skip(name, fn.length ? (t, done) => void fn(done) : async (t) => void await (fn as () => Promise<void>)());
bunit.if = (condition) => condition ? bunit : () => void 0;
bunit.skipIf = (condition) => condition ? bunit.skip : bunit;
bunit.each = (table: any) => {
    return (title: string, test: AnyFunction) => jesteach(table).it(title, test);
};

const testModule = {
    // This entire function is overall very hacky and little tested for now.
    // Maybe it would be better to just monkeypatch the relevant JS apis instead?
    setSystemTime(now?: Date | number) {
        if (process.platform === 'linux') {
            const sudo = ''; // process.getuid?.() === 0 ? '' : 'sudo ';
            if (typeof now === 'undefined') {
                exec(`${sudo}timedatectl set-ntp true`);
                return this;
            }
            //? Doesn't work on non-systemd distros, nor WSL by default...
            exec(
                `${sudo}timedatectl set-ntp false && ` +
                `${sudo}date -s "${dateFormat(now, "UTC:mm/dd/yyyy HH:MM:ss")}" --utc && ` +
                `${sudo}hwclock -w --utc`
            );
        } else if (process.platform === 'win32') {
            const Win32DateFormat = (() => {
                try {
                    const stdout = chp.execSync('date');
                    return stdout.toString('utf8').match(/Enter the new date: \((.+)\)/)?.[1] ?? 'dd-mm-yy';
                } catch (e) {
                    const err = e as { stdout: Buffer; };
                    return err.stdout.toString('utf8').match(/Enter the new date: \((.+)\)/)?.[1] ?? 'dd-mm-yy';
                }
            })();
            if (typeof now === 'undefined') {
                // TODO: How to reset system time on Windows? Below might work but is messy and needs admin...
                /* net stop w32time
                   w32tm /unregister
                   w32tm /register
                   net start w32time
                   w32tm /resync */
                return this;
            }
            exec(
                `date ${dateFormat(now, Win32DateFormat)} && ` +
                `time ${dateFormat(now, "HH:MM:ss")}`
            );
        } else throw new Error(`Unsupported platform for setSystemTime: ${process.platform}`); // TODO: How to set system time on MacOS? Can't test for now :(
        return this;
    },
    spyOn<T extends object, K extends keyof T>(obj: T, methodOrPropertyValue: K): Mock<T[K] extends AnyFunction ? T[K] : never> {
        const mock = jestmock.spyOn(
            obj,
            // @ts-expect-error jest has a really convoluted type for this that isnt worth trying to replicate
            methodOrPropertyValue,
        ) as jestmock.Spied<any>;
        // @ts-expect-error same reason as above
        return mock;
    },
    beforeAll(fn) {
        nodetest.before(fn.length ? (s, done) => void fn(done) : (s) => void (fn as () => void)());
    },
    beforeEach(fn) {
        nodetest.beforeEach(fn.length ? (s, done) => void fn(done) : (s) => void (fn as () => void)());
    },
    afterAll(fn) {
        nodetest.after(fn.length ? (s, done) => void fn(done) : (s) => void (fn as () => void)());
    },
    afterEach(fn) {
        nodetest.afterEach(fn.length ? (s, done) => void fn(done) : (s) => void (fn as () => void)());
    },
    mock: bunmock,
    jest: {
        restoreAllMocks() {
            bunmock.restore();
        },
        fn(func) {
            return jestmock.fn(func);
        },
    },
    describe: bundescribe,
    test: bunit,
    it: bunit,
    expect: bunExpect, // TODO: this is not fully compatible, needs finer grained implementation
} satisfies typeof import('bun:test');

export const setSystemTime = testModule.setSystemTime;
export const spyOn = testModule.spyOn;
export const beforeAll = testModule.beforeAll;
export const beforeEach = testModule.beforeEach;
export const afterAll = testModule.afterAll;
export const afterEach = testModule.afterEach;
export const mock = testModule.mock;
export const jest = testModule.jest;
export const describe = testModule.describe;
export const test = testModule.test;
export const it = testModule.it;
export const expect = testModule.expect;
export default testModule;
