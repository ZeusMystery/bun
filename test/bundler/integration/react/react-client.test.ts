import path from "path";
import { describe, test, expect } from "bun:test";
import { bunExe } from "../../../harness";

const modes = [
  //
  { label: "base" },
  { label: "minify-all", args: ["--minify"] },
  { label: "minify-syntax", args: ["--minify-syntax"] },
  { label: "minify-whitespace", args: ["--minify-whitespace"] },
  { label: "sourcemaps", args: ["--minify", "--sourcemap=external"] },
];
const nodeEnvs = ["development", "production"];
const combinations = nodeEnvs.flatMap(nodeEnv => modes.map(mode => ({ options: mode, nodeEnv })));

describe("bundler integration, react client", () => {
  for (const {
    options: { label, args },
    nodeEnv,
  } of combinations) {
    test(label + ", NODE_ENV=" + nodeEnv, async () => {
      const out = path.join(import.meta.dir, "dist/client/" + label + "-" + nodeEnv);
      const cmd = [
        bunExe(),
        "build",
        ...(args ?? []),
        "--outdir=" + out,
        "--splitting",
        path.join(import.meta.dir, "index.jsx"),
      ];
      console.log(cmd.join(" "));
      const x = Bun.spawnSync(cmd, {
        cwd: import.meta.dir,
        env: nodeEnv ? { NODE_ENV: nodeEnv } : undefined,
      });
      if (x.exitCode !== 0) {
        console.error(x.stderr.toString());
        throw new Error("Failed to build");
      }
      const proc = Bun.spawnSync(["node", path.join(import.meta.dir, "puppeteer.mjs"), out], {
        cwd: path.join(import.meta.dir),
      });
      if (!proc.success) {
        console.error(proc.stderr.toString());
        console.error(proc.stdout.toString());
        expect(proc.exitCode).toBe(0);
      }
      const output = JSON.parse(proc.stdout.toString("utf-8"));
      expect(output.logs).toMatchSnapshot("Browser console logs");
      expect(output.domSnapshots).toMatchSnapshot("DOM Snapshots");
    });
  }
});
