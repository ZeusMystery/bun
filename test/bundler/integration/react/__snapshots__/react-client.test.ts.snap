// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`bundler integration, react client base, NODE_ENV=development: Browser console logs 1`] = `
[
  "%cDownload the React DevTools for a better development experience: https://reactjs.org/link/react-devtools font-weight:bold",
  "render in <App/> http://localhost:3000/index.js",
  "render in <Loading/>",
  "useEffect1 in <App/>",
  "useEffect2 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
]
`;

exports[`bundler integration, react client base, NODE_ENV=development: DOM Snapshots 1`] = `
[
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 0</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div>Loading...</div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 1</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 2</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
]
`;

exports[`bundler integration, react client minify-all, NODE_ENV=development: Browser console logs 1`] = `
[
  "%cDownload the React DevTools for a better development experience: https://reactjs.org/link/react-devtools font-weight:bold",
  "render in <App/> http://localhost:3000/index.js",
  "render in <Loading/>",
  "useEffect1 in <App/>",
  "useEffect2 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
]
`;

exports[`bundler integration, react client minify-all, NODE_ENV=development: DOM Snapshots 1`] = `
[
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 0</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div>Loading...</div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 1</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 2</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
]
`;

exports[`bundler integration, react client minify-syntax, NODE_ENV=development: Browser console logs 1`] = `
[
  "%cDownload the React DevTools for a better development experience: https://reactjs.org/link/react-devtools font-weight:bold",
  "render in <App/> http://localhost:3000/index.js",
  "render in <Loading/>",
  "useEffect1 in <App/>",
  "useEffect2 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
]
`;

exports[`bundler integration, react client minify-syntax, NODE_ENV=development: DOM Snapshots 1`] = `
[
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 0</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div>Loading...</div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 1</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 2</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
]
`;

exports[`bundler integration, react client minify-whitespace, NODE_ENV=development: Browser console logs 1`] = `
[
  "%cDownload the React DevTools for a better development experience: https://reactjs.org/link/react-devtools font-weight:bold",
  "render in <App/> http://localhost:3000/index.js",
  "render in <Loading/>",
  "useEffect1 in <App/>",
  "useEffect2 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
]
`;

exports[`bundler integration, react client minify-whitespace, NODE_ENV=development: DOM Snapshots 1`] = `
[
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 0</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div>Loading...</div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 1</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 2</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
]
`;

exports[`bundler integration, react client sourcemaps, NODE_ENV=development: Browser console logs 1`] = `
[
  "%cDownload the React DevTools for a better development experience: https://reactjs.org/link/react-devtools font-weight:bold",
  "render in <App/> http://localhost:3000/index.js",
  "render in <Loading/>",
  "useEffect1 in <App/>",
  "useEffect2 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
  "-- testing marker --",
  "render in <App/> http://localhost:3000/index.js",
  "useEffect1 cleanup in <App/>",
  "useEffect1 in <App/>",
]
`;

exports[`bundler integration, react client sourcemaps, NODE_ENV=development: DOM Snapshots 1`] = `
[
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 0</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div>Loading...</div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 1</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
  
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>React App Integration Test</title>
</head>
<body>
  <div id="root"><div><h1 class="title">Hello, world!</h1><p class="class">React Application</p><div>static string</div><p class="count">Count: 2</p><button>Increment</button><p class="same-file">Component in same file</p><p class="another-file">Component in another file 1</p><div class="list"><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div><div>Item {}</div></div><div><p class="lazy">Lazy loaded chunk: http://localhost:3000/Lazy.js</p><code></code></div></div></div>
  <script src="/index.js" type="module"></script>


</body></html>"
,
]
`;
